app/components/TaskList.tsx

```diff
@@ -21,14 +22,32 @@ const TaskList: React.FC<Props> = ({ projectRevision }) => {
     `,
     projectRevision
   );
+  const router = useRouter();
+
+  const determineStepHighlight = (step) => {
+    if (!router || !router.pathname) return null;
+    if (router.pathname.includes(step)) {
+      return "highlight";
+    }
+    return null;
+  };
+
+  const determineSummaryHighlight = () => {
+    if (!router || !router.pathname) return null;
+    return `${
+      router.pathname === "/cif/project-revision/[projectRevision]"
+        ? "highlight"
+        : null
+    } bordered`;
+  };
```

Comment:
```suggestion
  const currentStep = useMemo(() => {
    if (!router || !router.pathname) return null;
    if (router.pathname === "/cif/project-revision/[projectRevision]") return "summary";
    const routeParts = router.pathname.split("/");
    return routeParts[routeParts.length - 1]
  }, [router.pathname])
```

Then below you can simply check `currentStep === "overview"`, `currentStep === "managers"`, etc.
This will have the advantage to only be called once per page.

Using `getProjectRevisionPageRoute(id)` instead of hardcoding `"/cif/project-revision/[projectRevision]"` above would also be better, in which case you'd want to use `router.asPath` instead of `router.pathname`